homework18/src/ba/bitcamp/homework18/task01/ViewComplains.java

```diff
@@ -0,0 +1,87 @@
+package ba.bitcamp.homework18.task01;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.ScrollPaneConstants;
+
+public class ViewComplains extends JFrame {
+
+	private static final long serialVersionUID = -3746179192521483602L;
+	SQLConn connection = null;
+	Statement statement = null;
+	ResultSet result = null;
+	ArrayList<Complain> complains;
+
+	private JTextArea area = new JTextArea(17, 22);
+	private JPanel panel = new JPanel();
+	private JScrollPane pane = new JScrollPane(area);
+
+	public ViewComplains(SQLConn connection) {
+		
+		this.connection = connection;
+		
+		add(panel);
+		panel.add(pane);
+		pane.setHorizontalScrollBarPolicy(ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
+		area.setWrapStyleWord(true);
+		area.setLineWrap(true);
+		area.setEditable(false);
+
+		printComplains(allComplains());
+
+		setSize(300, 320);
+		setDefaultCloseOperation(EXIT_ON_CLOSE);
+		setLocationRelativeTo(null);
+		setVisible(true);
+		setResizable(false);
+	}
+
+	/*
+	 * This method returns all complains from database and puts them to
+	 * arraylist as objects "Complain". So we can manipulate them as Objects,
+	 * and print them.
+	 */
+	private ArrayList<Complain> allComplains() {
+		complains = new ArrayList<Complain>();
+		try {
+			statement = connection.getConn().createStatement();
+			String query = "select * from complain";
```

Comment:
Fula koja te po meni moze nekada kostati nekada nekoliko sati debug-inga, a to je koristenje ovog pristupa result.getInt(1). Moje pitanje da bih ti objasnio sta se desava probaj dodati kolonu na poziciju 1 u tabeli i zamisli da si ovu logiku koristio na jedno 100 mjesta :). Znaci morao bi proci svih 100 mjesta i pomjeriti sve za jednu poziciju.
Znaci moj ti je savjet uvijek koristi name pristup tj, result.getInt("ID") jer onda iako dodas bilo gdje kolonu samo gdje ti treba ta nova kolona dodas za nju logiku citanja.